admission/hooks/mutate_httpproxy.go

```diff
@@ -64,6 +65,97 @@ func (m *contourHTTPProxyMutator) Handle(ctx context.Context, req admission.Requ
 	if err != nil {
 		return admission.Errored(http.StatusInternalServerError, err)
 	}
+	return admission.Allowed("ok")
+}
+
+func getHTTPProxyPolicy(policies []HTTPProxyPolicy, key string) (HTTPProxyPolicy, bool) {
+	for _, v := range policies {
+		if v.Name == key {
+			return v, true
+		}
+	}
+	return HTTPProxyPolicy{}, false
+}
+
+func setHTTPProxyPolicyField(hp *unstructured.Unstructured, policy HTTPProxyPolicy) error {
+	routes, ok, err := unstructured.NestedSlice(hp.UnstructuredContent(), "spec", "routes")
+	if err != nil {
+		return err
+	}
+	if !ok {
+		return nil
+	}
+	for _, route := range routes {
+		ipPoliciesOrg, _, err := unstructured.NestedSlice(route.(map[string]interface{}), "ipAllowPolicy")
+		if err != nil {
+			return err
+		}
+		ipPolicies := []HTTPProxyIPFilterPolicy{}
+		for _, v := range ipPoliciesOrg {
+			jsonStr, err := json.Marshal(v)
+			if err != nil {
+				return err
+			}
+			ipPolicy := HTTPProxyIPFilterPolicy{}
+			if err := json.Unmarshal(jsonStr, &ipPolicy); err != nil {
+				return err
+			}
+			ipPolicies = append(ipPolicies, ipPolicy)
+		}
+		ipPolicies = append(ipPolicies, policy.IpAllowPolicy...)
+
+		ipPoliciesMap := map[HTTPProxyIPFilterPolicy]struct{}{}
+		for _, v := range ipPolicies {
+			ipPoliciesMap[v] = struct{}{}
+		}
+		uniqIpPolicies := []interface{}{}
+		for k := range ipPoliciesMap {
+			jsonStr, err := json.Marshal(k)
+			if err != nil {
+				return err
+			}
+			ipPolicy := map[string]interface{}{}
+			if err := json.Unmarshal(jsonStr, &ipPolicy); err != nil {
+				return err
+			}
+			uniqIpPolicies = append(uniqIpPolicies, ipPolicy)
+		}
+		err = unstructured.SetNestedSlice(route.(map[string]interface{}), uniqIpPolicies, "ipAllowPolicy")
```

Comment:
If I write `uniqIpPolicies = append(uniqIpPolicies, k)` here,
`unstructured.SetNestedSlice` messages `cannot deep copy hooks.HTTPProxyIPFilterPolicy` and panic occurs.
Panic is avoided by converting to `map[string]interface{}` via json.